src/main/java/com/qunite/api/web/ResponseEntityControllerAdvice.java

```diff
@@ -9,36 +9,53 @@
 import com.qunite.api.web.dto.ExceptionResponse;
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
+import org.springframework.dao.DataIntegrityViolationException;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.RestControllerAdvice;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
 
 @RestControllerAdvice
 public class ResponseEntityControllerAdvice {
   @ExceptionHandler({
-      QueueNotFoundException.class,
-      UserNotFoundException.class,
-      EntryNotFoundException.class})
-  public ResponseEntity<ExceptionResponse> handleNotFound(RuntimeException exception) {
-    return ResponseEntity.status(HttpStatus.NOT_FOUND)
+      UserAlreadyExistsException.class,
+      IllegalArgumentException.class,
+      HttpMessageNotReadableException.class,
+      MethodArgumentTypeMismatchException.class})
+  public ResponseEntity<ExceptionResponse> handleBadRequest(RuntimeException exception) {
+    return ResponseEntity.status(HttpStatus.BAD_REQUEST)
         .body(exceptionResponse(exception.getMessage()));
   }
 
-  @ExceptionHandler(UserAlreadyExistsException.class)
-  public ResponseEntity<ExceptionResponse> handleBadRequest(RuntimeException exception) {
-    return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+  @ExceptionHandler(JWTDecodeException.class)
+  public ResponseEntity<ExceptionResponse> handleUnauthorized(RuntimeException exception) {
+    return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
         .body(exceptionResponse(exception.getMessage()));
   }
 
-  @ExceptionHandler({
-      ForbiddenAccessException.class,
-      JWTDecodeException.class})
+  @ExceptionHandler(ForbiddenAccessException.class)
   public ResponseEntity<ExceptionResponse> handleForbidden(RuntimeException exception) {
     return ResponseEntity.status(HttpStatus.FORBIDDEN)
         .body(exceptionResponse(exception.getMessage()));
   }
 
+  @ExceptionHandler({
+      QueueNotFoundException.class,
+      UserNotFoundException.class,
```

Comment:
replace with EntityNotFoundException?

Reply:
> replace with EntityNotFoundException?

no need, separate custom exceptions were introduced to display more detailed information

Reply:
What exactly detailed information they introduce? They are always handled in the same way and i see no different situation. As an alternative we can inherit all this exceptions from Entity Not found